asg-passes/src/dotify/program.rs

```diff
@@ -0,0 +1,134 @@
+            ("Annotations", format!("{:?}", input.annotations)), //Note: Debug seems to work
+        ];
+
+        Dotifier::add_span_info(&mut labels, &input.span);
+
+        let start_idx = self.add_or_get_node(input.id, "Function".to_string(), labels);
+
+        for (i, Fixed(end_idx)) in arguments.iter().enumerate() {
+            self.add_edge(start_idx, *end_idx, format!("variable_{:}", i), "olive");
+        }
+
+        let Fixed(end_idx) = body;
+        self.add_edge(start_idx, end_idx, "body".to_string(), "black");
+
+        if let Some(circuit) = input.circuit.get() {
+            self.edges.push((input.id, circuit.id, "circuit".to_string(), "green"))
+        }
+
+        Fixed(start_idx)
+    }
+
+    fn reduce_circuit_member(&mut self, input: &CircuitMember<'a>, function: Option<M>) -> M {
+        if let CircuitMember::Variable(typ) = input {
+            let id = self.context.get_id();
+            let labels = vec![("NodeID", id.to_string()), ("Type", typ.to_string())];
+
+            let start_idx = self.add_or_get_node(id, "Variable".to_string(), labels);
+            Fixed(start_idx)
+        } else {
+            function.unwrap() // If circuit member is a function, monoidal director always visits it
+        }
+    }
+
+    fn reduce_circuit(&mut self, input: &'a Circuit<'a>, members: Vec<M>) -> M {
+        let mut labels = vec![
+            ("NodeID", input.id.to_string()),
+            ("Name", input.name.borrow().name.to_string()),
+        ];
+
+        Dotifier::add_span_info(&mut labels, &input.span);
+
+        let start_idx = self.add_or_get_node(input.id, "Circuit".to_string(), labels);
+
+        for (i, Fixed(end_idx)) in members.iter().enumerate() {
+            self.add_edge(start_idx, *end_idx, format!("member_{:}", i), "black");
+        }
+
+        Fixed(start_idx)
+    }
+
+    fn reduce_alias(&mut self, input: &'a Alias<'a>) -> M {
+        let labels = vec![("NodeID", input.id.to_string())];
+        let start_idx = self.add_or_get_node(input.id, "Alias".to_string(), labels);
+
+        Fixed(start_idx)
+    }
+
+    fn reduce_program(
+        &mut self,
+        input: &Program,
+        imported_modules: Vec<M>,
+        aliases: Vec<M>,
+        functions: Vec<M>,
+        global_consts: Vec<M>,
+        circuits: Vec<M>,
+    ) -> M {
+        let labels = vec![("NodeID", input.id.to_string()), ("Name", input.name.clone())];
+
+        let start_idx = self.add_or_get_node(input.id, "Program".to_string(), labels);
+
+        for (i, Fixed(end_idx)) in imported_modules.iter().enumerate() {
+            self.add_edge(start_idx, *end_idx, format!("import_{:}", i), "orange");
+        }
+
+        for (i, Fixed(end_idx)) in aliases.iter().enumerate() {
+            self.add_edge(start_idx, *end_idx, format!("alias_{:}", i), "pink");
+        }
+
+        for (i, Fixed(end_idx)) in functions.iter().enumerate() {
+            self.add_edge(start_idx, *end_idx, format!("function_{:}", i), "black");
```

Comment:
The [X11 color scheme](https://graphviz.org/doc/info/colors.html) has a lot of options. It doesn't seem practical to make enums for all of them, or select a subset (in case someone wants to modify the rendering).

Reply:
I mean only the colors *you are using* in the code. That also makes it easier to see what is actually used in a centralized place.